cluster/context.go

```diff
@@ -0,0 +1,37 @@
+// This file is part of MinIO Kubernetes Cloud
+// Copyright (c) 2019 MinIO, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package cluster
+
+import (
+	"context"
+	"database/sql"
+)
+
+// An application wide context that holds the a transaction, in case anything
+// goes wrong during the business logic execution, database objects can be
+// rolled back.
+type Context struct {
+	*sql.Tx
+	Main *context.Context
```

Comment:
I kind of want to have a traditional context attached to our application specific context so we can still handle cancellations and timeouts, this was the only way I could think of it, would you recommend I rather hold the instance instead of a pointer?